src/test/kotlin/app/CodeLongevityTest.kt

```diff
@@ -0,0 +1,248 @@
+
+        assertEquals(expectedCommit, actualLine.commit, "$messsage commit")
+        assertEquals(expectedFile, actualLine.file, "$messsage file name")
+        assertEquals(expectedLineNum, actualLine.line, "$messsage line num")
+    }
+
+    /**
+     * Assert function to test CodeLine object.
+     */
+    fun assertCodeLine(lineText: String,
+                       fromCommit: RevCommit, fromFile: String, fromLineNum: Int,
+                       toCommit: RevCommit, toFile: String, toLineNum: Int,
+                       actualLine: CodeLine) {
+
+        assertRevCommitLine(fromCommit, fromFile, fromLineNum, actualLine.from,
+                            "'$lineText' from_commit");
+        assertRevCommitLine(toCommit, toFile, toLineNum, actualLine.to,
+                            "'$lineText' to_commit");
+        assertEquals(lineText, actualLine.text, "line text")
+    }
+
+    given("'test group #1'") {
+        val testRepoPath = "../testrepo1"
+        val testRepo = TestRepo("../testrepo1")
+        val fileName = "test1.txt"
+
+        // t1: initial insertion
+        testRepo.newFile(fileName, listOf("line1", "line2"))
+        val rev1 = testRepo.commit("inital commit")
+        val lines1 = CodeLongevity(
+            LocalRepo(testRepoPath), Repo(), MockApi(), testRepo.git).compute()
+
+        it("t1: initial insertion") {
+            assertEquals(2, lines1.size)
+            assertCodeLine("line1",
+                           rev1, fileName, 0,
+                           rev1, fileName, 0,
+                           lines1[0])
+            assertCodeLine("line2",
+                           rev1, fileName, 1,
+                           rev1, fileName, 1,
+                           lines1[1])
+        }
+
+        // t2: subsequent insertion
+        testRepo.insertLines(fileName, 1, listOf("line in the middle"))
+        val rev2 = testRepo.commit("insert line")
+        val lines2 = CodeLongevity(
+            LocalRepo(testRepoPath), Repo(), MockApi(), testRepo.git).compute()
+
+        it("t2: subsequent insertion") {
+            assertEquals(3, lines2.size)
+            assertCodeLine("line in the middle",
+                           rev2, fileName, 1,
+                           rev2, fileName, 1,
+                           lines2[0])
+            assertCodeLine("line1",
+                           rev1, fileName, 0,
+                           rev2, fileName, 0,
+                           lines2[1])
+            assertCodeLine("line2",
+                           rev1, fileName, 1,
+                           rev2, fileName, 2,
+                           lines2[2])
+        }
+
+        // t3: subsequent deletion
+        testRepo.deleteLines(fileName, 2, 2)
+        val rev3 = testRepo.commit("delete line")
+        val lines3 = CodeLongevity(LocalRepo(testRepoPath), Repo(),
+                                   MockApi(), testRepo.git).compute()
+
+        it("'t3: subsequent deletion'") {
```

Comment:
Quotes?

Reply:
you mean single quotes? it just looks nicer when errors are reported in the console

Reply:
In first tests no single quotes.